macro/src/derive.rs

```diff
@@ -313,7 +310,7 @@ impl Driver<'_> {
             let pattern = variant.pat();
             return Ok(quote_spanned!(self.span() =>
                 #pattern => {
-                    #binding .jderef().upcast().execute_with(jvm)
```

Comment:
when would one call `execute_with` vs. `do_jni`? Both are on the `JvmOp` trait AFAICT and it seems like sometimes we call one or the other?